### PR TITLE
Add AnnounceTask and StatTask tests

### PR DESCRIPTION
**Pull request by gaius-qi**
Signed-off-by: Gaius <gaius.qi@gmail.com>

## Description
- Add `AnnounceTask` and `StatTask` tests.
- Refactor dferror message.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: scheduler/service/service.go [hunk 1]
```diff
@@ -79,9 +79,9 @@ func (s *Service) RegisterPeerTask(ctx context.Context, req *rpcscheduler.PeerTa
 	// Register task and trigger cdn download task
 	task, err := s.registerTask(ctx, req)
 	if err != nil {
-		dferr := dferrors.New(base.Code_SchedTaskStatusError, "register task is fail")
-		logger.Errorf("peer %s register is failed: %v", req.PeerId, err)
-		return nil, dferr
+		msg := fmt.Sprintf("peer %s register is failed: %v", req.PeerId, err)
```
Comment (cndoit18) [inline scheduler/service/service.go:82]:
```suggestion
		msg := fmt.Sprintf("peer %s register is failed: %w", req.PeerId, err)
```

File: scheduler/service/service.go [hunk 2]
```diff
@@ -96,9 +96,9 @@ func (s *Service) RegisterPeerTask(ctx context.Context, req *rpcscheduler.PeerTa
 			peer.Log.Info("task size scope is tiny and return piece content directly")
 			if len(task.DirectPiece) > 0 && int64(len(task.DirectPiece)) == task.ContentLength.Load() {
 				if err := peer.FSM.Event(resource.PeerEventRegisterTiny); err != nil {
-					dferr := dferrors.New(base.Code_SchedError, err.Error())
-					peer.Log.Errorf("peer %s register is failed: %v", req.PeerId, err)
-					return nil, dferr
+					msg := fmt.Sprintf("peer %s register is failed: %v", req.PeerId, err)
```
Comment (cndoit18) [inline scheduler/service/service.go:99]: msg := fmt.Sprintf("peer %s register is failed: %w", req.PeerId, err)

**Review by cndoit18 — APPROVED** [review]
LGTM

**Comment by eryugey** [comment]
LGTM! Thanks for the tests!

**Review by lucy-cl — APPROVED** [review]

**Review by jim3ma — APPROVED** [review]
LGTM

**Pull request merged**